pkg/virtual/syncer/builder/build.go

```diff
@@ -82,23 +83,36 @@ func BuildVirtualWorkspace(
 			withoutRootPathPrefix := strings.TrimPrefix(urlPath, rootPathPrefix)
 
 			// Incoming requests to this virtual workspace will look like:
-			//  /services/syncer/root:org:ws/<sync-target-name>/clusters/*/api/v1/configmaps
+			//  /services/syncer/root:org:ws/<sync-target-name>/<sync-target-uid>/clusters/*/api/v1/configmaps
 			//                  └───────────────────────────┐
 			// Where the withoutRootPathPrefix starts here: ┘
-			parts := strings.SplitN(withoutRootPathPrefix, "/", 3)
-			if len(parts) < 2 || parts[0] == "" || parts[1] == "" {
+			parts := strings.SplitN(withoutRootPathPrefix, "/", 4)
+			if len(parts) < 3 || parts[0] == "" || parts[1] == "" || parts[2] == "" {
 				return
 			}
+			workspace := parts[0]
 			workloadCusterName := parts[1]
+			syncTargetUID := parts[2]
 			apiDomainKey := dynamiccontext.APIDomainKey(clusters.ToClusterAwareKey(logicalcluster.New(parts[0]), workloadCusterName))
 
+			syncTarget, err := kcpClusterClient.Cluster(logicalcluster.New(workspace)).WorkloadV1alpha1().SyncTargets().Get(context.TODO(), workloadCusterName, metav1.GetOptions{})
```

Comment:
must be informer based.

Reply:
done.

Reply:
logs are still too verbose for my taste.